app/src/main/java/org/kiwix/kiwixmobile/zim_manager/fileselect_view/effects/DeleteFiles.kt

```diff
@@ -40,18 +40,20 @@ data class DeleteFiles(private val booksOnDiskListItem: List<BookOnDisk>) :
 
   override fun invokeWith(activity: AppCompatActivity) {
     (activity as ZimManageActivity).cachedComponent.inject(this)
-    booksOnDiskListItem.forEach {
-      dialogShower.show(DeleteZim(it), {
-        if (deleteSpecificZimFile(it)) {
-          if (it.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
+
+    val name = booksOnDiskListItems.joinToString(separator = "\n") { it.book.title }
+
+    dialogShower.show(DeleteZims(name), {
+      val booksDeleted = booksOnDiskListItems.fold(true) { acc, book ->
+        acc && deleteSpecificZimFile(book).also {
+          if (it && book.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
             zimReaderContainer.setZimFile(null)
           }
-          activity.toast(R.string.delete_specific_zim_toast)
-        } else {
-          activity.toast(R.string.delete_zim_failed)
         }
-      })
-    }
+      }
+
+      activity.toast(if (booksDeleted) R.string.delete_zims_toast else R.string.delete_zim_failed)
```

Comment:
excellent looks great, only way it could maybe read better would be an extension function so this entire body becomes
```
activity.toast(
if(booksOnDiskListItems.deleteAll()) R.string.delete_zims_toast
else R.string.delete_zim_failed)
```
however unnecessary.

This has my approval, I'll be back in a bit to merge unless you want to do the extension function

Reply:
@macgills if I make an extension function, it will be like the below:

```kotlin
private fun List<BookOnDisk>.deleteAll(): Boolean {
    val name = this.joinToString(separator = "\n") { it.book.title }

    dialogShower.show(DeleteZims(name), {
      val booksDeleted = this.fold(true) { acc, book ->
        acc && deleteSpecificZimFile(book).also {
          if (it && book.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
            zimReaderContainer.setZimFile(null)
          }
        }
      }
    })
  }
```

So I will need to change booksDeleted into a `var` and increase its scope so that I can return the required value successfully. 
Should I proceed this way?

Reply:
```
fold(true) { acc, book ->
        acc && deleteSpecificZimFile(book).also {
          if (it && book.file.canonicalPath == zimReaderContainer.zimCanonicalPath) {
            zimReaderContainer.setZimFile(null)
          }
        }
```
this is the part I think should be in an extension function, if you simply return this fold you have the value you want, no `booksDeleted` variable at all

Reply:
@macgills I should've noticed that :sweat_smile: 
I've made the changes.